### PR TITLE
Implement retry logic in html5 provider

### PR DESCRIPTION
**Pull request by robwalch**
### This PR will...
Implement retry logic in html5 provider

### Why is this Pull Request needed?
Safari makes a lot of manifest requests after loading the page and player, re-requesting manifests almost every time the tab focus or system audio context changes. Since delivery of content can be compromised for one of these requests (bad status, cached result for expired signed url) we need to build in some resiliency for handling these errors.

These changes build on top of retry logic for decode errors (code 3) https://github.com/jwplayer/jwplayer/pull/3401 by adding a retry for not supported error (code 2). This means we can recover before or after playback has started. It also adds support for 3 tries, and resets the retry count after 2 consecutive timeupdate events.

### Are there any Pull Requests open in other repos which need to be merged with this?
Updates to this HLS proxy script allow for testing against a stream that alternates between returning a healthy stream and one which contains child manifests that 403:
https://github.com/jwplayer/jwplayer-commercial/pull/6733

#### Addresses Issue(s):
JW8-9327

### CODE REVIEW

**Comment by johnBartos** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  🛠 There are modified src files, but no test changes. Add tests if you're able to.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/js/providers/html5.js [hunk 1]
```diff
@@ -86,6 +93,10 @@ function VideoProvider(_playerId, _playerConfig, mediaElement) {
         },
 
         timeupdate() {
+            if (_this.currentTime) {
+                // Reset error retries after concurrent timeupdate events
+                _this.retries = 0;
```
Comment (johnBartos) [inline src/js/providers/html5.js]: Reset on init

File: src/js/providers/html5.js [hunk 2]
```diff
@@ -194,14 +205,19 @@ function VideoProvider(_playerId, _playerConfig, mediaElement) {
             const error = video.error;
             const errorCode = (error && error.code) || -1;
 
-            if (errorCode === 3 && _this.currentTime !== -1 && OS.iOS) {
-                // Workaround iOS bug https://bugs.webkit.org/show_bug.cgi?id=195452
+            if (errorCode === 3 || errorCode === 4 && _this.retries < 3) {
```
Comment (waxidiotic) [inline src/js/providers/html5.js]: So we would trigger this warning if `errorCode === 3` OR `errorCode === 4 && _this.retries < 3`?

File: src/js/providers/html5.js (hunk 1 above)
Comment (robwalch) [reply src/js/providers/html5.js]: done

File: src/js/providers/html5.js (hunk 2 above)
Comment (robwalch) [reply src/js/providers/html5.js]: Nice catch. Fixed!

**Pull request merged**